ledger/src/blockstore.rs

```diff
@@ -1366,7 +1412,17 @@ impl Blockstore {
                         String
                     )
                 );
-            return false;
+
+            // If consumed <= shred_index, then We haven't gotten all the shreds up to `shred_index` yet.
+            // In this case, there's a chance hat since `shred_index` < the current last index, this shred
+            // index is necessary to complete the entry for replay. Worst case it's the wrong version,
+            // replay marks the slot as dead. However, if we don't insert here, then there's a chance we never
+            // get all the shreds < the current last index, never replay, and make no progress (for instance if
+            // a leader sends an additional detached "last index" shred with a very high index, but none of the
```

Comment:
Isn't it still possible to get stuck if a node received a very high shred index for an unrepairable block and they already consumed more shreds than the *real* block? It feels to me that the block should be marked dead if it ever gets a last shred that's less than its current last shred. Then your dupe handling could repair the full slot if it detects a dupe?